kong/cmd/vault.lua

```diff
@@ -97,6 +97,19 @@ The available commands are:
 Options:
  -c,--conf    (optional string)  configuration file
  -p,--prefix  (optional string)  override prefix directory
+
+Caveats:
+  This command doesn't work in DBless mode if <reference> uses vaults entity.
+  It doesn't affect the normal use of vault in kong instance though.
+
+  The reason is in DBless mode the vaults entity is stored in LMDB
+  which is implemented by a Nginx C module. However Everytime `resty` cli
+  (which is relied on by `kong` cli) runs it creates a temporary `nginx.conf`
+  which doesn't contain the lmdb-related directives.
+
+  A workaround is manully adding these directives prefixed with `lmdb_`
+  in `nginx.conf` into the `resty` script or via the `--main-conf` option
+  of `resty`.
```

Comment:
The problem is described in some detail, but I think the workaround that is suggested is too incomplete to be of use to a normal user (i.e. the list of parameters that need to be is not provided).

I would suggest that either the bug is fixed or, if that is not feasible, the documentation just indicates that vault references are not supported.  It would be even better if the `get` command just refused to work when a vault reference is provided in dbless mode.

Reply:
Thanks for the suggestion. Trying to fix this.

Reply:
i would also prefer bug to be fixed. It may be a source of multiple issues.

Reply:
@bungle This one is ready. Could you help review this?